sdk/java/pulumi/src/main/java/com/pulumi/serialization/internal/ResourcePackages.java

```diff
@@ -158,7 +158,21 @@ private static boolean excludePackages(ClassInfo c) {
                 && !c.getPackageName().equals("net.bytebuddy")
                 && !c.getPackageName().startsWith("net.bytebuddy.")
                 && !c.getPackageName().startsWith(Resource.class.getPackageName())
-                && !c.getPackageName().startsWith(EngineGrpc.class.getPackageName());
+                && !c.getPackageName().startsWith(EngineGrpc.class.getPackageName())
+                && !excludePackagesFromProperties(c);
+    }
+
+    private static boolean excludePackagesFromProperties(ClassInfo c) {
+        String packages = System.getProperty("ADDITIONAL_EXCLUDES");
```

Comment:
Should we pick a better name than `ADDITIONAL_EXCLUDES` for this? Is there a standard idiomatic naming convention that most other Java libraries typically use for system properties? For example, wondering if this should be something like `pulumi.resourcepackages.excludes` or `pulumi.classloader.excludes` or something like that?

Reply:
Yes this is a very valid point, I think pulumi.resourcepackages.excludes fit the purpose very well.